src/inference/include/openvino/runtime/core.hpp

```diff
@@ -55,6 +55,10 @@ class OPENVINO_RUNTIME_API Core {
      */
     explicit Core(const std::string& xml_config_file = {});
 
+    /** @brief Constructs an OpenVINO Core instance with devices and their plugins description. */
+    template <class Path, std::enable_if_t<std::is_same_v<Path, std::filesystem::path>>* = nullptr>
+    explicit Core(const Path& xml_config_file) : Core(xml_config_file.string()) {}
```

Comment:
There is no reason to use SFINAE here. Common overloading is enough. @praasz please confirm.

```suggestion
    explicit Core(const std::filesystem::path& xml_config_file) : Core(xml_config_file.string()) {}
```